.github/actions/run_local_pytest/action.yml

```diff
@@ -0,0 +1,48 @@
+name: 'Install Local Python dependencies'
+description: 'Install Local Python dependencies'
+inputs:
+  job_type:          {default: '', type: string, description: Specifies which steps should be executed}
+  build_type:        {default: 'release', type: string, description: Specifies whether the build that we are testing is release or debug}
+  threads:           {default: '1', type: string, description: Specifies the number of threads that will be used for tests }
+  fast_tests_only:   {default: '0', type: string, description: Specifies whether we want to run only the fast tests or not }
+  other_params:      {default: '', type: string, description: Other params that should be passed to the python command }
+runs:
+  using: "composite"
+  steps:
+    - name: Install local dependencies with pip
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+
+        python >build_tooling/requirements-analysis-workflow.txt <<END
+        from configparser import ConfigParser
+        cf = ConfigParser()
+        cf.read("setup.cfg");
+
+        # setup_requires is not needed
+        print(cf['options']['setup_requires'])
+        print(cf['options']['install_requires'])
```

Comment:
Have you tried `pip install -v .`? You shouldn't need to do any of these by hand.

Also, `setup_requires` and `install_requires` are only used by `pip` internally and shouldn't appear in the "runtime" environment. I.e. a real user doing `pip install arcticdb` won't have those dependencies.

Reply:
wouldn't `pip install -v .` trigger a rebuild? the point is to only install packages that we depend on, without recompiling because, presumably, we have already compiled and with specific definitions/presets

I agree about `setup_requires` but I think that we need `install_requires`, those are the packages that we depend on, no?

Reply:
Okay thanks, I will experiment with this